doc/developer/design/20220203_connectors.md

```diff
@@ -0,0 +1,211 @@
+  ('keepalives' '=' keepalives ',')?
+  (postgres_ssl_options)?
+postgres_ssl_options ::=
+  ('sslmode' '=' ( 'disable' | 'allow' | 'prefer' | 'require' | 'verify-ca' | 'verify-full' ) ',')?
+  ('sslcert' '=' ssl_certificate_location ',')?
+  ('sslkey' '=' ssl_key_location (',' 'sslpassword' '=' ssl_key_password )? ',')?
+  ('sslsni' '=' ssl_sni ',')?
+```
+### AWS connector
+```ebnf
+create_connector_aws ::=
+  ( ('access_key_id' '=' access_key_id ',' 'secret_access_key' '=' secret_access_key (',' 'token' '=' token)?) | 'profile' '=' profile ) 
+  ('role_arn' '=' role_arn)?
+  ('region' '=' region)?
+```
+
+## Alter Table complexities
+Catalog support for these relations would require more extensive changes to properly disentangle the shareable portions of a source description from the elements which are expected to be unique. This may be a desirable evolution of these structures anyhow to organize them more modularly, primarily along boundaries of responsibility such as authentication.
+
+This would need to be done with some care to avoid leaking unnecessary implementation details into the grammar, while still avoiding `WITH options` wherever possible. The initial connector specification would provide a solid baseline of moving all Source but not implementation specific details out of `WITH options` and into the grammar. 
+
+Creating a more universal abstraction of authentication for sources is valuable beyond just the potential for Connector usage, since several source types support distinctly different auth modes when using cloud hosted versions;  Kafka and Postgres in AWS supporting IAM authentication are the most obvious example where the current representation of the source effectively disallows representing this auth method in any way.
+
+#### CASCADE specific complexities
+
+While allowing ALTER to CASCADE to all sources which reference the connector makes certain operational tasks for users easier, such as changing rotating authentication certificates or passwords, updating broker host names, or even completely changing auth methods. Implementation brings with it a significant amount of essential complexity to ensure robust, correct behavior even with eventually consistent updates of running sources.
+
+Before considering the transactional consistency required to communicate changes from ADAPTER to STORAGE, the first added complexity comes in ensuring that only ALTERs which actually change the value in a specific source are propagated which requires re-evaluating the entire set of connectors involved in specifying a source.s 
+
+Upsert would be simplest way to extend this; however, this introduces distributed coordination requirements in any mode of operation. By making Upsert semantically fallible STORAGE can incrementally support live ALTER on a source by source and property by property basis.
+
+- Eventually consistent changes would involve Adapter and Storage participating in 2 phase commit
```

Comment:
This doesn't look like 2PC to me? 2PC would be the ADAPTER initiating the UPSERT with STORAGE in the first phase, then in the second phase telling STORAGE to actually commit the UPSERT.  It's not clear to me what happens in this sequence diagram, btw, if the ADAPTER dies after its UPSERT request has succeeded but before it learns of that response.

As long as we allow for eventual consistency, I don't think we need 2PC, though? If we make "upsert source" an infallible operation (which it basically is if you manage to construct a `SourceConnector`), then the ADAPTER can just tell the client "yes" and update STORAGE at its leisure.

Reply:
Yeah, this terminology wasn't the right choice. This whole section is pretty irrelevant based on some of the other feedback so I will scrap it.